core/src/main/scala/org/apache/spark/util/collection/ExternalSorter.scala

```diff
@@ -748,6 +759,12 @@ private[spark] class ExternalSorter[K, V, C](
 
     context.taskMetrics.memoryBytesSpilled += memoryBytesSpilled
     context.taskMetrics.diskBytesSpilled += diskBytesSpilled
+    context.taskMetrics.shuffleWriteMetrics.filter(_ => bypassMergeSort).foreach { m =>
```

Comment:
Lies :stuck_out_tongue_winking_eye:

I don't think I've ever written a `.filter(_ => ...` though I will admit guilt for merging all but one instance in the codebase...